apps/allocations/app/components/Card/Budget.js

```diff
@@ -0,0 +1,213 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+import { BigNumber } from 'bignumber.js'
+
+import {
+  Card,
+  ContextMenu,
+  ContextMenuItem,
+  IconEdit,
+  IconPlus,
+  IconProhibited,
+  IconView,
+  ProgressBar,
+  Text,
+  useTheme,
+} from '@aragon/ui'
+import { displayCurrency } from '../../utils/helpers'
```

Comment:
We shouldn't need to do this here, but instead separate the logic into our `app-state-reducer` to keep our components dumb and specialized into layout and style

Reply:
Got it. I'll do it like that next time. But this time I'll leave it to you so that my commits are still testable in a standalone way.

Reply:
I will probably modify this part to take advantage of the appStateReducer transformations for this data